models/points.go

```diff
@@ -367,6 +367,20 @@ func scanKey(buf []byte, i int) (int, []byte, error) {
 			pos += copy(b[pos:], v)
 		}
 
+		// Check again for duplicate tags now that the tags are sorted.
+		for j := 0; j < commas-1; j++ {
+			// get the left and right tags
+			_, left := scanTo(buf[indices[j]:], 0, '=')
+			_, right := scanTo(buf[indices[j+1]:], 0, '=')
+
+			// If the tags are equal, then there are duplicate tags, and we should abort.
+			// If the tags are not sorted, this pass may not find duplicate tags and we
+			// need to do a more exhaustive search later.
+			if bytes.Equal(left, right) {
+				return i, b, fmt.Errorf("duplicate tags")
+			}
+		}
```

Comment:
Is this `duplicate tags` error being tested? It seems like the test cases below would be caught in the error block above (during the first scan).